src/com/mendix/widget/slider/components/Slider.ts

```diff
@@ -0,0 +1,134 @@
+import { Component, DOM, createElement } from "react";
+
+import * as classNames from "classnames";
+import * as RcSlider from "rc-slider";
+
+import "../ui/Slider.css";
+import "rc-slider/dist/rc-slider.css";
+
+import { Alert } from "./Alert";
+
+export interface SliderProps {
+    hasError?: boolean;
+    value?: number;
+    noOfMarkers?: number;
+    maxValue?: number | null;
+    minValue?: number | null;
+    validationMessage?: string;
+    onClick?: (value: number) => void;
+    onChange?: (value: number) => void;
+    stepValue?: number;
+    tooltipText?: string | null;
+    disabled: boolean;
+    decimalPlaces?: number;
+}
+
+interface Marks {
+    [key: number]: string | number | {
+        style: HTMLStyleElement,
+        label: string
+    };
+}
+
+export class Slider extends Component<SliderProps, {}> {
+    static defaultProps: SliderProps = {
+        disabled: false,
+        maxValue: 100,
+        minValue: 0,
+        noOfMarkers: 2,
+        tooltipText: "{1}"
+    };
+
+    constructor(props: SliderProps) {
+        super(props);
+
+        this.getTooltipText = this.getTooltipText.bind(this);
+    }
+
+    render() {
+        const alertMessage = this.validateSettings(this.props)
+            || this.validateValues(this.props) || this.props.validationMessage;
+
+        return DOM.div({ className: classNames("widget-slider", { "has-error": !!alertMessage }) },
+            createElement(RcSlider, {
+                disabled: !!this.validateSettings(this.props) || this.props.disabled,
+                included: true,
+                marks: this.calculateMarks(this.props),
+                max: this.props.maxValue,
+                min: this.props.minValue,
+                onAfterChange: this.props.onClick,
+                onChange: this.props.onChange,
+                pushable: false,
+                range: false,
+                step: this.props.stepValue ? this.props.stepValue : null,
+                tipFormatter: this.props.tooltipText ? this.getTooltipText : null,
```

Comment:
Unlike the step value, here we check if there is some text, we call a function getTooltiptext else its null